lib/logstash/inputs/udp.rb

```diff
@@ -74,11 +72,13 @@ def udp_listener(output_queue)
       Thread.new { inputworker(i) }
     end
 
-    while true
+    while !stop?
       #collect datagram message and add to queue
       payload, client = @udp.recvfrom(@buffer_size)
```

Comment:
I wonder why not using here the non blocking call and checking for payload == String.empty? and IO::WaitReadable. This would be basically giving you the same effect as this without the cumbersome need to close the socket to exit the loop. If you aim to implement a semi blocking behaviour you could also use IO.select.

What do you think?